crates/viewer/re_space_view_spatial/src/visualizers/assets3d.rs

```diff
@@ -87,28 +81,26 @@ impl Asset3DVisualizer {
             if let Some(mesh) = mesh {
                 re_tracing::profile_scope!("mesh instances");
 
-                let world_from_pose = ent_context.world_from_entity
-                    * data
-                        .transform
-                        .map_or(glam::Affine3A::IDENTITY, |t| t.0.into());
-
-                instances.extend(mesh.mesh_instances.iter().map(move |mesh_instance| {
-                    let pose_from_mesh = mesh_instance.world_from_mesh;
-                    let world_from_mesh = world_from_pose * pose_from_mesh;
-
-                    MeshInstance {
-                        gpu_mesh: mesh_instance.gpu_mesh.clone(),
-                        world_from_mesh,
-                        outline_mask_ids,
-                        picking_layer_id: picking_layer_id_from_instance_path_hash(
-                            picking_instance_hash,
-                        ),
-                        ..Default::default()
-                    }
-                }));
-
-                self.0
-                    .add_bounding_box(entity_path.hash(), mesh.bbox(), world_from_pose);
+                // Let's draw the mesh once for every instance transform. Because why not!
```

Comment:
Nit: I like the behavior, but don't like the "Because why not" philosophy.

I'd prefer to see this linked to an issue about defining consistent archetype semantics for when components within an archetype act as:
- Batch-joined. I.e. Arch[index] = (C1[index], C2[index], ...). Example: Points
- Instance-joined. I.e. Arch[0] = (C1[:], C2[:], ...). Example: Mesh3D
- Product-joined. I.e. Arch[i1, i2] = (C[i1], C[i2]).

Mesh3D is of course a hybrid of Instance-joined components Vertex / Indices with Product-joined component of LeafTransform3D.

Reply:
improving comment notes pointing this out.